T/TVM/build_tarballs.jl

```diff
@@ -1,69 +1,76 @@
+    ArchiveSource("https://dlcdn.apache.org/tvm/tvm-v$(version)/apache-tvm-src-v$(version).tar.gz", "2da001bf847636b32fc7a34f864abf01a46c69aaef0ff37cfbfbcc2eb5b0fce4"),
+    ArchiveSource("https://github.com/phracker/MacOSX-SDKs/releases/download/10.15/MacOSX10.15.sdk.tar.xz",
+        "2408d07df7f324d3beea818585a6d990ba99587c218a3969f924dfcc4de93b62"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-# Start by moving things into the proper places
-for submodule in dlpack dmlc-core rang HalideIR; do
-    rm -rf "${WORKSPACE}/srcdir/tvm/3rdparty/${submodule}"
-    mv "${WORKSPACE}/srcdir/${submodule}" "${WORKSPACE}/srcdir/tvm/3rdparty/${submodule}"
-done
+cd $WORKSPACE/srcdir/apache-tvm-src-v0.10.0/
+install_license LICENSE 
+mkdir build && cd build
 
-# Next, enter TVM and start building it
-cd $WORKSPACE/srcdir/tvm
+# setup LLVM_LIBS manually for non-Linux OS
+if [[ "$target" == *darwin* ]]; then
+    # Work around "'value' is unavailable"
+    export MACOSX_DEPLOYMENT_TARGET=10.15
+    # ...and install a newer SDK which supports `std::filesystem`
+    pushd $WORKSPACE/srcdir/MacOSX10.*.sdk
+    rm -rf /opt/${target}/${target}/sys-root/System
+    cp -ra usr/* "/opt/${target}/${target}/sys-root/usr/."
+    cp -ra System "/opt/${target}/${target}/sys-root/."
+    popd
 
-mkdir build && cd build
-cmake .. -DCMAKE_INSTALL_PREFIX="${prefix}" \
-         -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}" \
-         -DCMAKE_
-make -j${nproc}
-make install
+    export LLVM_LIBS="${libdir}/libLLVM-14.0.dylib"
+else
+    if [[ "$target" == *mingw* ]]; then
+        export LLVM_LIBS="${bindir}/libLLVM-14jl.dll"
+    fi
+fi
+
+cmake .. -DCMAKE_INSTALL_PREFIX=${prefix} \
+         -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+         -DCMAKE_BUILD_TYPE=Release \
+         -DUSE_LLVM=ON \
+         -DLLVM_LIBS=${LLVM_LIBS} \
+         -G Ninja
+
+ninja
+ninja install
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = [
+    Platform("i686", "linux"; libc="glibc"),
     Platform("x86_64", "linux"; libc="glibc"),
-
-    # Fails with can't find `execinfo.h`
-    #Platform("x86_64", "linux"; libc="musl"),
     Platform("x86_64", "macos"),
 
     # dll import woes
-    #Platform("x86_64", "windows"),
+    # Platform("x86_64", "windows"),
 ]
+platforms = expand_cxxstring_abis(platforms)
 
 # The products that we will ensure are always built
-products = Product[
+products = [
     LibraryProduct("libtvm", :libtvm),
-    LibraryProduct("libtvm_topi", :libtvm_topi),
     LibraryProduct("libtvm_runtime", :libtvm_runtime),
-    LibraryProduct("libnnvm_compiler", :libnnvm_compiler),
 ]
 
 # Dependencies that must be installed before this package can be built
-dependencies = [
-    "Zlib_jll",
-    "LLVM_jll",
+dependencies = Dependency[
+    Dependency(PackageSpec(name="Zlib_jll")),
+    Dependency(PackageSpec(name="LLVM_jll")),
```

Comment:
Do you need full llvm or just libllvm (which is a much smaller package)?  Also, if this links dynamically to libllvm at runtime, you're locking into a single llvm.  You want to see what https://github.com/JuliaPackaging/Yggdrasil/blob/79007df5f1de4a7b58119e87b5a7ad0ee6e984f0/L/LLVMExtra/build_tarballs.jl does.